sympy/polys/polytools.py

```diff
@@ -3773,6 +3774,178 @@ def nth_power_roots_poly(f, n):
+            numbers.
+        MultivariatePolynomialError
+            If the polynomial is not univariate.
+        PolynomialError
+            If the polynomial is a monomial, since in such a case it has no
+            nonzero roots.
+
+        Notes
+        =====
+
+        The domain of the polynomial must be a subring of the complex numbers.
+        We use the `Cauchy bound`_.
+
+        .. _Cauchy bound: https://en.wikipedia.org/wiki/Geometrical_properties_of_polynomial_roots#Lagrange's_and_Cauchy's_bounds
+
+        """
+        dom = f.get_domain()
+        if dom.is_FiniteField or not dom.is_Numerical:
+            raise DomainError("Cannot compute root bounds over %s" % dom)
+        if f.is_multivariate:
+            raise MultivariatePolynomialError(
+                "Must be a univariate polynomial")
+        if f.is_monomial:
+            raise PolynomialError("Monomial has no nonzero roots")
+
+        coeffs = f.all_coeffs()
+        while coeffs[-1] == 0:
+            coeffs.pop()
+        lc, tc = coeffs[0], coeffs[-1]
+        ub = 1 + max(abs(n / lc) for n in coeffs[1:])
+        lb_inv = 1 + max(abs(n / tc) for n in coeffs[:-1])
+
+        return 1/lb_inv, ub
+
+    def root_separation_lower_bound(f):
+        """
+        Compute a lower bound on the minimum distance between any two
+        roots of this polynomial.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly
+        >>> from sympy.abc import x
+        >>> Poly(x**2 - 2).root_separation_lower_bound()
+        1
+        >>> (x * Poly(x**2 - 2)).root_separation_lower_bound()
+        12/17
+
+        Returns
+        =======
+
+        :py:class:`~.Expr`
+            A non-negative lower bound which will be positive if all the roots
+            are distinct, and rational if the coefficients of the polynomial
+            are rational.
+
+        Raises
+        ======
+
+        DomainError
+            If the domain of the polynomial is not a subring of the complex
+            numbers.
+        MultivariatePolynomialError
+            If the polynomial is not univariate.
+
+        """
+        dom = f.get_domain()
+        if dom.is_FiniteField or not dom.is_Numerical:
+            raise DomainError("Cannot compute root bounds over %s" % dom)
+        if f.is_multivariate:
+            raise MultivariatePolynomialError(
+                "Must be a univariate polynomial")
+        if f.is_monomial:
+            return Integer(0)
+
+        x = f.gen
+        y = Dummy('y')
+        g = f.subs(x, x + y)
+        h = Poly(resultant(f, g, gens=[x, y]), domain=f.domain)
+        # If f has roots ri, the roots of h are all the ri - rj.
+
+        if h.is_monomial:
+            # In this case all roots of f are the same.
+            return Integer(0)
+
+        lb, _ = h.root_bounds()
+
+        # Since h is an even function, can improve bound with square root.
+        if lb.is_rational:
```

Comment:
This should probably be `is_Rational` rather than `is_rational`. Note the difference:
```python
In [4]: a = S(3)

In [5]: b = Symbol('b', rational=True)

In [6]: a
Out[6]: 3

In [7]: b
Out[7]: b

In [8]: a.is_rational
Out[8]: True

In [9]: a.is_Rational
Out[9]: True

In [10]: b.is_rational
Out[10]: True

In [11]: b.is_Rational
Out[11]: False
```
The test `is_Rational` is structural: is this an explicit rational number. The test `is_rational` is a *semantic* query through the assumptions system:
https://docs.sympy.org/latest/guides/assumptions.html

The polys module is mostly concerned with structural considerations rather than semantic in this sense.

Reply:
Ah, thanks for the explanation. i was wondering about this.